packages/graph/hash_graph/lib/graph/src/api/rest/utoipa_typedef/subgraph/edges.rs

```diff
@@ -56,7 +77,42 @@ impl From<crate::subgraph::edges::Edges> for Edges {
             knowledge_graph: KnowledgeGraphRootedEdges(edges.knowledge_graph.into_iter().fold(
                 HashMap::new(),
                 |mut map, (id, edges)| {
-                    let edges = edges.into_iter().collect();
+                    let edges = edges.into_iter().map(|edge| {
+                        match edge {
+                            crate::subgraph::edges::KnowledgeGraphOutwardEdges::ToOntology(edge) => {
+                                KnowledgeGraphOutwardEdges::ToOntology(edge)
+                            }
+                            crate::subgraph::edges::KnowledgeGraphOutwardEdges::ToKnowledgeGraph(
+                                edge,
+                            ) => {
+                                let earliest_timestamp = if edge.reversed {
```

Comment:
I was struggling to wrap my head around it as I wasn't sure this branch logic is correct. I see now (at least I think) it is. Can you add a comment explaining something like this

We always need the earliest version of the _link_ entity, and there are four permutations
* hasLeftEntity (link to left)
* hasLeftEntity reversed  (left to link)
* hasRightEntity (link to right)
* hasRightEntity reversed (right to link)

If it's reversed the link is on the right endpoint, otherwise it's on the left

Reply:
Yes, it's a little bit too hard to get the branch logic. I added a better description to this location: 14938b548c5dea70a4e51bcd8f3a94cc3e00b5c8